corehq/apps/users/management/commands/clone_web_users.py

```diff
@@ -0,0 +1,183 @@
+            domain=domain_membership.domain,
+            timezone=domain_membership.timezone,
+            override_global_tz=domain_membership.override_global_tz,
+            role_id=domain_membership.role_id,
+            location_id=domain_membership.location_id,
+            assigned_location_ids=domain_membership.assigned_location_ids,
+            program_id=domain_membership.program_id
+        )
+        to_user.domain_memberships.append(copied_membership)
+        to_user.domains.append(copied_membership.domain)
+
+    to_user.save()
+
+
+def transfer_exports(from_user, to_user):
+    for domain in from_user.domains:
+        key = [domain]
+        for export in _get_export_instance(ExportInstance, key):
+            if export.owner_id == from_user.get_id:
+                export.owner_id = to_user.get_id
+                export.save()
+
+
+def transfer_scheduled_reports(from_user, to_user_id):
+    for domain in from_user.domains:
+        for scheduled_report in ReportNotification.by_domain_and_owner(domain, from_user._id):
+            scheduled_report.owner_id = to_user_id
+            scheduled_report.save()
+
+
+def transfer_saved_reports(from_user, to_user):
+    for domain in from_user.domains:
+        for saved_report in ReportConfig.by_domain_and_owner(domain, from_user.get_id):
+            saved_report.owner_id = to_user.get_id
+            saved_report.save()
+
+
+def copy_user_fields(from_user, to_user):
+    # DjangoUserMixin fields
+    # username, email and password have already been set
+    # do not copy last_login and date_joined
+    to_user.first_name = from_user.first_name
+    to_user.last_name = from_user.last_name
+    to_user.is_staff = from_user.is_staff
+    to_user.is_active = from_user.is_active
+    to_user.is_superuser = from_user.is_superuser
+
+    # CouchUser fields
+    # ignoring device_ids, last_device, created_on, last_modified
+    to_user.devices = from_user.devices
+    to_user.phone_numbers = from_user.phone_numbers
+    to_user.status = from_user.status
+    to_user.language = from_user.language
+    to_user.subscribed_to_commcare_users = from_user.subscribed_to_commcare_users
+    to_user.announcements_seen = from_user.announcements_seen
+    to_user.user_data = from_user.user_data
+    to_user.update_metadata(from_user.metadata)
+    to_user.set_location(from_user.location_id)
+    to_user.assigned_location_ids = from_user.assigned_location_ids
+    to_user.has_built_app = from_user.has_built_app
+    to_user.analytics_enabled = from_user.analytics_enabled
+
+    # Web User fields
+    to_user.program_id = from_user.program_id
+    to_user.fcm_device_token = from_user.fcm_device_token
+    to_user.atypical_user = from_user.atypical_user
+
+    # EulaMixin field
+    to_user.eulas = from_user.eulas
+
+    return to_user
+
+
+def send_deprecation_email(old_user, new_user):
+    context = {
+        'greeting': _("Dear %s,") % new_user.first_name if new_user.first_name else _("Hello,"),
```

Comment:
nit `_("Dear {first_name},").format(first_name=new_user.first_name)` instead of `%` is preferred